### PR TITLE
Add skip flag to skip Esti tests

### PR DESCRIPTION
**Pull request by idanovo**

### CODE REVIEW

**Review by nopcoder — CHANGES_REQUESTED** [review]
Can we use the same syntax as '-run' - if go will add this capability they will keep the same syntax probably and it will be easy to migrate.

File: esti/main_test.go [hunk 1]
```diff
@@ -309,6 +313,13 @@ func TestMain(m *testing.M) {
...
 	defer func() { _ = server.s.Close() }()
 
+	testsToSkipMap = arrayFlagsToMap(testsToSkip)
 	logger.Info("Setup succeeded, running the tests")
 	os.Exit(m.Run())
 }
+
+func SkipTestIfAskedTo(t *testing.T) {
+	if testsToSkipMap[t.Name()] {
+		t.Skip(fmt.Printf("Skip on test: %s", t.Name()))
```
Comment (nopcoder) [inline esti/main_test.go:323]: just `SkipNow()` as we don't give any good explanation :)

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: esti/main_test.go [hunk 2]
```diff
@@ -308,7 +312,14 @@ func TestMain(m *testing.M) {
 	}
 
 	defer func() { _ = server.s.Close() }()
+	testsToSkipRegex, _ = regexp.Compile(*testsToSkip)
```
[the remark this answers is not in the document]
Comment (idanovo) [reply esti/main_test.go:315]: In case we have an error I should fail the run or just run all the tests?

File: esti/main_test.go [hunk 3]
```diff
@@ -308,7 +312,17 @@ func TestMain(m *testing.M) {
 	}
 
 	defer func() { _ = server.s.Close() }()
+	testsToSkipRegex, err = regexp.Compile(*testsToSkip)
+	if err != nil {
+		log.Fatal(err)
```
Comment (nopcoder) [inline esti/main_test.go:317]:
```suggestion
		log.Fatalf("Skip pattern '%s' failed to compile:", *testsToSkip, err)
```

File: esti/main_test.go [hunk 4]
```diff
@@ -308,7 +312,17 @@ func TestMain(m *testing.M) {
 	}
 
 	defer func() { _ = server.s.Close() }()
+	testsToSkipRegex, err = regexp.Compile(*testsToSkip)
```
Comment (nopcoder) [inline esti/main_test.go:315]: if *testsToSkip != "" {
}

File: esti/main_test.go [hunk 5]
```diff
@@ -308,7 +312,17 @@ func TestMain(m *testing.M) {
...
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	logger.Info("Setup succeeded, running the tests")
 	os.Exit(m.Run())
 }
+
+func SkipTestIfAskedTo(t *testing.T) {
+	if testsToSkipRegex.String() != "" && testsToSkipRegex.MatchString(t.Name()) {
```
Comment (nopcoder) [inline esti/main_test.go:325]: testsToSkipRegex
```suggestion
	if testsToSkipRegex != nil && testsToSkipRegex.MatchString(t.Name()) {
```

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by nopcoder — APPROVED** [review]
LGTM! hope we will have it as part of the next Go release

**Pull request merged**